fabric-rendering-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/BuiltinItemRendererRegistryImpl.java

```diff
@@ -56,8 +51,21 @@ public void register(ItemConvertible item, BuiltinItemRenderer renderer) {
 		register(item.asItem(), renderer);
 	}
 
+	@Override
+	public void register(ItemConvertible item, DynamicItemRenderer renderer) {
+		Objects.requireNonNull(item, "item is null");
+		Objects.requireNonNull(item.asItem(), "item is null");
+		Objects.requireNonNull(renderer, "renderer is null");
+
+		if (RENDERERS.containsKey(item.asItem())) {
```

Comment:
```suggestion
		if (RENDERERS.put(item.asItem(), renderer) != null) {
```

Reply:
This will still overwrite the existing entry if the modder decides to catch any exceptions produced by this method, which is incorrect behaviour IMO. If that exception is thrown, there should be no side effects to the registry itself.

Reply:
then `putIfAbsent`